src/Microsoft.OData.Client/BulkUpdateSaveResult.cs

```diff
@@ -119,13 +168,27 @@ internal void BuildDescriptorGraph<T>(IEnumerable<Descriptor> descriptors, bool
                 {
                     if (descriptor is EntityDescriptor entityDescriptor)
                     {
-                        if (entityDescriptor.ParentEntity != null && 
-                            entityDescriptor.ParentEntity.Equals(topLevelDescriptor.Entity) && 
+                        if (entityDescriptor.ParentEntity != null &&
+                            entityDescriptor.ParentEntity.Equals(topLevelDescriptor.Entity) &&
                             !bulkUpdateGraph.Contains(entityDescriptor))
                         {
                             bulkUpdateGraph.AddRelatedDescriptor(topLevelDescriptor, entityDescriptor);
                             this.BuildDescriptorGraph(descriptors, false, entityDescriptor.Entity);
                         }
+                        else
+                        {
+                            // We check if the link representation of the descriptor exists in the entity tracker and it is unchanged. 
+                            LinkDescriptor linkDescriptor = this.RequestInfo.EntityTracker.Links.Where(
+                                a => a.Source == topLevelDescriptor.Entity && 
+                                this.RequestInfo.EntityTracker.GetEntityDescriptor(a.Target) == entityDescriptor &&
+                                a.State == EntityStates.Unchanged).FirstOrDefault();
+                            if (linkDescriptor != null)
+                            {
+                                bulkUpdateGraph.AddRelatedDescriptor(topLevelDescriptor, entityDescriptor);
+                                this.linkDescriptors.Add(entityDescriptor, linkDescriptor);

```

Comment:
What if we have multiple different link descriptors linking to the same target entity descriptors? Then would previous entries with the same entity descriptor be overwritten? I think you should add a test case for this scenario and verify whether it works.

Reply:
@ElizabethOkerio See my comment here about scenario where the nested level is greater than 1 and confirm that the logic here works

https://github.com/OData/odata.net/pull/2627#discussion_r1165045106